drivers/timer/stm32_lptim_timer.c

```diff
@@ -61,11 +65,41 @@ static const struct device *const clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_N
  *    0xFFFF / (LSE freq (32768Hz) / 128)
  */
 
-static uint32_t lptim_clock_freq = KHZ(32);
-static int32_t lptim_time_base;
+#if (CONFIG_STM32_LPTIM_CLOCK_LSI)
+static uint32_t lptim_clock_freq = KHZ(32); /* Force STM32_SRC_LSI */
 
-/* The prescaler given by the DTS and to apply to the lptim_clock_freq */
-#define LPTIM_CLOCK_RATIO DT_PROP(DT_DRV_INST(0), st_prescaler)
+/* Kconfig defines the clock source as LSI : check coherency with DTS */
+#if (DT_INST_CLOCKS_CELL_BY_IDX(0, 1, bus) != STM32_SRC_LSI)
+#warning CONFIG_STM32_LPTIM_CLOCK_LSI requires STM32_SRC_LSI (DTS)
+#endif /* STM32_SRC_LSI */
+
+/* Kconfig should set the adapted CONFIG_SYS_CLOCK_TICKS_PER_SEC */
+#if !defined(CONFIG_STM32_LPTIM_TICK_FREQ_RATIO_OVERRIDE)
+#if ((CONFIG_SYS_CLOCK_TICKS_PER_SEC > 32000/LPTIM_CLOCK_RATIO) || \
+	(CONFIG_SYS_CLOCK_TICKS_PER_SEC > 4000))
+#warning It is recommended to set the tick freq to 32000 / prescaler
```

Comment:
Warning is generated on a "lower than" condition, so warning message should be coherent with that.
For instance:
```
It is recommended to set the tick freq value lower than 32000 / prescaler
```


But in the end, I think it would be easier to understand if we have a warning about "tick per clock cycle":
(if possible, express the check in tick per clock cycle:)
```
#if (CONFIG_SYS_CLOCK_TICKS_PER_SEC * LPTIM_CLOCK_RATIO / CONFIG_STM32_LPTIM_CLOCK_LSE > 1)
```
Then write the warning using the same notion
```
#warning Configuration requests more than 1 tick per clock cycle. This is not valid.
```

Reply:
Another possibility is to complete the soc/arm/st_stm32/common/Kconfig.defconfig.series with 
```
if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSE
config SYS_CLOCK_TICKS_PER_SEC
	default 4096 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),clock-prescaler)" < 16
	default 2048 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),clock-prescaler)" = 16
	default 1024 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),clock-prescaler)" = 32
```